### PR TITLE
TECH: add git hooks

### PR DESCRIPTION
**Pull request by eakurnikov**

### CODE REVIEW

**Review by AzamatCherchesov — DISMISSED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -36,3 +36,19 @@ val detektAll = tasks.register<Detekt>("detektAll") {
         html.enabled = false
     }
 }
+
+tasks.register<Exec>("installGitHooks") {
```
Comment (VladislavSumin) [inline build.gradle.kts:40]: nit: move to separate convention file

File: build.gradle.kts [hunk 2]
```diff
@@ -36,3 +36,19 @@ val detektAll = tasks.register<Detekt>("detektAll") {
...
+    description = "Install local git hooks"
+    group = "Build Setup"
+
+    commandLine("chmod")
+    args("-R", "u+x", ".githooks")
```
Comment (VladislavSumin) [inline build.gradle.kts:45]: git remembers file attributes anyway, why do we need to explicitly set them manually?

File: .githooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+# https://git-scm.com/docs/githooks#_pre_commit
+set -eo pipefail
+
+currentBranch="$(git rev-parse --abbrev-ref HEAD)"
+branchNamePattern="^(ISSUE-\d+(-[^\W_]+)*|TECH(-[^\W_]+)+|HEAD)$"
+
+if ! echo "$currentBranch" | grep "$branchNamePattern" -qE; then
+  echo >&2 "ERROR: pre-commit git hook failed."
+  echo >&2 "Branch name must contain issue id and match the pattern: \"ISSUE-1234{-additional-description}\"."
```
Comment (VladislavSumin) [inline .githooks/pre-commit:10]: nit: issue or tech, not only issue

File: .githooks/commit-msg [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+set -eo pipefail
+
+gitMessageFile="$1"
+gitMessage=$(cat $gitMessageFile)
+gitMessagePattern="^(Revert \")?(ISSUE-\d+|TECH):\s\w+.*"
+currentBranch="$(git rev-parse --abbrev-ref HEAD)"
+
+if ! grep -qE "$gitMessagePattern" "$gitMessageFile"; then
+  echo >&2 "ERROR: commit-msg git hook failed."
+  echo >&2 "Commit message must start with issue id and match the pattern: \"ISSUE-1234: git commit message\"."
```
Comment (VladislavSumin) [inline .githooks/commit-msg:12]: nit: issue or tech, not only issue

**Review by RuslanMingaliev — APPROVED** [review]

File: .githooks/commit-msg [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# https://git-scm.com/docs/githooks#_commit_msg
+set -eo pipefail
+
+gitMessageFile="$1"
+gitMessage=$(cat $gitMessageFile)
+gitMessagePattern="^(Revert \")?(ISSUE-\d+|TECH):\s\w+.*"
```
[the remark this answers is not in the document]
Comment (eakurnikov) [reply .githooks/commit-msg:7]: As far as I see on the board we use issues for anything, for features, for bugs and so on. If you are doing some small fix and don't want to create a dedicated issue for it, you can simply use TECH.

**Pull request merged**